invenio_rdm_records/cli.py

```diff
@@ -278,3 +280,176 @@ def rebuild_index():
     affs_service.rebuild_index(identity=system_identity)
 
     click.secho("Reindexed records and vocabularies!", fg="green")
+
+
+# CUSTOM FIELDS
+
+
+@click.group()
+def custom_fields():
+    """InvenioRDM custom fields commands."""
+
+
+@custom_fields.group()
+def records():
+    """InvenioRDM custom fields records commands."""
+
+
+# helper functions
+def _prepare_mapping(fields_names, available_fields):
+    """Prepare ES mapping properties for each field."""
+    available_fields = {field.name: field for field in available_fields}
+    fields = []
+    if fields_names:
+        for field_name in fields_names:
+            fields.append(available_fields.get(field_name))
+    else:
+        fields = available_fields.values()
+
+    properties = {}
+    for field in fields:
+        properties[f"custom_fields.{field.name}"] = field.mapping
+
+    return properties
+
+
+def _exists(field_name, index):
+    """Check if a field exists in `index`'s mapping."""
+    mapping = list(index.get_mapping().values())[0]["mappings"]
+
+    parts = field_name.split(".")
+    for part in parts:
+        mapping = mapping["properties"]  # here to avoid last field access
+        if part not in mapping.keys():
+            return False
+        mapping = mapping[part]
+
+    return True
+
+
+@records.command("create")
+@click.option(
+    "-f",
+    "--field-name",
+    type=str,
+    required=False,
+    multiple=True,
+    help="A custom field name to create. If not provided, all custom fields will be created.",
+)
+@with_appcontext
+def create_records_custom_field(field_name):
+    """Creates one or all custom fields for records.
+
+    $ invenio custom-fields records create [field].
+    """
+    available_fields = current_app.config.get("RDM_CUSTOM_FIELDS", [])
+    if not available_fields:
+        click.secho("No custom fields were configured. Exiting...", fg="green")
+        return
+
+    click.secho("Creating all custom fields...", fg="green")
+    # multiple=True makes it an iterable
+    properties = _prepare_mapping(field_name, available_fields)
+
+    try:
+        record_index = current_rdm_records.records_service.config.record_cls.index
+        draft_index = current_rdm_records.records_service.config.draft_cls.index
+
+        record_index.put_mapping(body={"properties": properties})
+        draft_index.put_mapping(body={"properties": properties})
+        click.secho("Created all custom fields!", fg="green")
+    except RequestError as e:
+        click.secho("An error occured while creating custom fields.", fg="red")
```

Comment:
What about suggesting what to check?
```suggestion
        click.secho("An error occured while creating search mappings for custom fields. Is the search service healthy?", fg="red")
```

Reply:
Not really needed because the following line will print the actual error (e.g. ES connection refused). If we add this comment, then it might be misleading (imagine this says healthy and then the error message something about duplicate or permissions)